### PR TITLE
Switch proxy server lifecycle management + refactoring

### PR DESCRIPTION
**Pull request by aslepakurov**
- proxy server is no decoupled from web application
- proxy mode can be switched
- refactoring

### CODE REVIEW

File: satellite/web_application.py [hunk 1]
```diff
@@ -29,6 +44,19 @@ def __init__(self,  master):
...
             (r"/flows/(?P<flow_id>[0-9a-f\-]+)/(?P<message>request|response)/content/(?P<content_view>[0-9a-zA-Z\-\_]+)(?:\.json)?", FlowContentView)
         ])
+        satellite_ctx.webapp = self
+
+    def start(self):
+        AsyncIOMainLoop().install()
+        iol = tornado.ioloop.IOLoop.current()
+        self.http_server = tornado.httpserver.HTTPServer(self)
+        self.http_server.listen(self.master.options.web_port, self.master.options.web_host)
+        web_url = "http://{}:{}/".format(self.master.options.web_host, self.master.options.web_port)
```
Comment (Stepan-VGS) [inline satellite/web_application.py:58]: Small amendment, I think it is more convenient to use this format: `web_url = f"http://{self.master.options.web_host}:{self.master.options.web_port}/"`

File: satellite/web_application.py [hunk 2]
```diff
@@ -29,6 +44,19 @@ def __init__(self,  master):
...
+        self.master.log.info(
+            "Web server listening at {}".format(web_url),
```
Comment (Stepan-VGS) [inline satellite/web_application.py:60]: The same here as above

**Review by Stepan-VGS — APPROVED** [review]

File: satellite/web_application.py (hunk 1 above)
Comment (aslepakurov) [reply satellite/web_application.py]: Fixed, good catch 👍

File: satellite/runner.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        # Make sure that we catch KeyboardInterrupts on Windows.
+        # https://stackoverflow.com/a/36925722/934719
+        if os.name == "nt":
+            async def wakeup():
+                while True:
+                    await asyncio.sleep(0.2)
+            asyncio.ensure_future(wakeup())
+
+        # master.run()
```
Comment (rozenbom) [inline satellite/runner.py:128]: Remove?

File: satellite/proxy.py [hunk 4]
```diff
@@ -1,8 +1,4 @@
-import tornado.httpserver
-import tornado.ioloop
-from tornado.platform.asyncio import AsyncIOMainLoop
-
-from mitmproxy import master
+# from mitmproxy import master
```
Comment (rozenbom) [inline satellite/proxy.py:1]: Remove?

File: satellite/proxy.py [hunk 5]
```diff
@@ -13,14 +9,15 @@
 from mitmproxy.addons import termstatus
 from mitmproxy.addons import eventstore
 from mitmproxy.tools.web import static_viewer, webaddons
+# from mitmproxy.master import Master
```
Comment (rozenbom) [inline satellite/proxy.py:12]: Remove?

**Review by rozenbom — APPROVED** [review]

**Pull request merged**